### PR TITLE
Added staff events

### PR DESCRIPTION
**Pull request by akulsharma1**
- added GET `/staff/shift/` for staff members to view all events they have a shift for

### CODE REVIEW

File: src/services/staff/staff-router.ts [hunk 1]
```diff
@@ -74,4 +77,34 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
...
+    }
+
+    try {
+        const data: StaffShift | null = await Models.StaffShift.findOne({ userId: payload.id });
+
+        if (!data) {
+            return next(new RouterError(StatusCode.ServerErrorInternal, "ShiftNotFound"));
+        }
+
+        const shifts: string[] = data.shifts;
```
Comment (AydanPirani) [inline src/services/staff/staff-router.ts]:
```suggestion
        const shiftIds: string[] = data.shifts;
```

File: src/services/staff/staff-router.ts [hunk 2]
```diff
@@ -74,4 +77,34 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
...
+
+        console.log(shifts);
```
Comment (AydanPirani) [inline src/services/staff/staff-router.ts]: Remove log line

File: src/services/staff/staff-router.ts [hunk 3]
```diff
@@ -74,4 +77,34 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
...
 });
 
+staffRouter.get("/shift/", strongJwtVerification, async (_: Request, res: Response, next: NextFunction) => {
+    const payload: JwtPayload | undefined = res.locals.payload as JwtPayload;
+
+    if (!hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "Forbidden"));
+    }
+
+    try {
```
Comment (AydanPirani) [inline src/services/staff/staff-router.ts]: No need for try-catches now :)

**Review by AydanPirani — COMMENTED** [review]
Few changes really quick

File: src/services/staff/staff-router.ts (hunk 3 above)
Comment (akulsharma1) [reply src/services/staff/staff-router.ts]: Does the code automatically recover if theres an error thrown? or is there some other handling which needs to be added?

Comment (AydanPirani) [reply src/services/staff/staff-router.ts]: yeah - eg a database call for type X becomes:

`const data: X | null = Models...`, and it's null if nothing is returned - basically it doesn't throw an error, but is a wrapper that returns null for no data found

**Review by AydanPirani — APPROVED** [review]
lgtm!

**Pull request merged**